src/Features/CSharp/Portable/IntroduceVariable/CSharpIntroduceLocalForExpressionCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.IntroduceVariable;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.IntroduceVariable
```

Comment:
Nit: Namespace name should probably match the added trait: `IntroduceLocalForExpression`. Please change for all added types if you change it here.

Reply:
I'm adding this feature to the existing IntroduceVariable feature-space.  It's just a special subcase of that feature.  But i didn't want to add hte code to that feature because... well... it's gotten pretty insanely complex and it's just so much easier to have the simple refactoring for the simple case :)